tukaan/windows/wm.py

```diff
@@ -0,0 +1,559 @@
+from __future__ import annotations
+
+import ctypes
+import functools
+import re
+import sys
+from enum import Enum
+from fractions import Fraction
+from typing import Callable, Sequence
+
+from tukaan._data import Position, Size
+from tukaan._images import Icon
+from tukaan._tcl import Tcl
+from tukaan._utils import windows_only
+from tukaan.colors import Color
+from tukaan.enums import Resizable, WindowBackdropType, WindowState, WindowType
+from tukaan.exceptions import TukaanTclError
+
+if sys.platform == "win32":
+    windows_build = sys.getwindowsversion().build
+
+
+class AccentPolicy(ctypes.Structure):
+    _fields_ = [
+        ("AccentState", ctypes.c_int),
+        ("AccentFlags", ctypes.c_int),
+        ("GradientColor", ctypes.c_uint),
+        ("AnimationId", ctypes.c_int),
+    ]
+
+
+class WindowCompositionAttributeData(ctypes.Structure):
+    _fields_ = [
+        ("Attribute", ctypes.c_int),
+        ("Data", ctypes.POINTER(ctypes.c_int)),
+        ("SizeOfData", ctypes.c_size_t),
+    ]
+
+
+class Margins(ctypes.Structure):
+    _fields_ = [
+        ("cxLeftWidth", ctypes.c_int),
+        ("cxRightWidth", ctypes.c_int),
+        ("cyTopHeight", ctypes.c_int),
+        ("cyBottomHeight", ctypes.c_int),
+    ]
+
+
+class DWMWA(Enum):
+    TRANSITIONS_FORCEDISABLED = 3
+    USE_IMMERSIVE_DARK_MODE = 20
+    BORDER_COLOR = 34
+    CAPTION_COLOR = 35
+    TEXT_COLOR = 36
+    SYSTEMBACKDROP_TYPE = 38  # Windows 11 build >= 22621
+    MICA_EFFECT = 1029  # Undocumented
+
+
+class DesktopWindowManager:
+    @staticmethod
+    def set_window_attr(name_or_hwnd: str | int, attr: DWMWA, value: int) -> None:
+        hwnd = (
+            name_or_hwnd
+            if isinstance(name_or_hwnd, int)
+            else int(Tcl.call(str, "wm", "frame", name_or_hwnd), 16)
+        )
+
+        c_value = ctypes.c_int(value)
+
+        ctypes.windll.dwmapi.DwmSetWindowAttribute(
+            hwnd,
+            attr.value,
+            ctypes.byref(c_value),
+            ctypes.sizeof(ctypes.c_int),
+        )
```

Comment:
I suspect something's wrong here with the window handle, because `border_color` and other don't work either.

Reply:
Seems like `wm frame` returns the child window, and not the toplevel. A simple `update idletasks` solves the problem.